components/slider.js

```diff
@@ -1,23 +1,63 @@
 import React from 'react'
 import PropTypes from 'prop-types'
 import classNames from 'classnames'
-import animateScrollTo from 'animated-scroll-to';
+import scrollToWithAnimation from 'scrollto-with-animation';
 import { Highlight } from '../components/highlights'
 import stylesheet from './slider.scss'
 import { ArrowButton } from './arrow-button';
 import { MediaList } from './icon';
 import { IndustriesMain } from './industries-main';
 
+const autoSliderTime = 15000;
+
 export class Slider extends React.Component {
   constructor(props) {
     super(props);
     this.state = {currentScroll: 0, isBeginning: true, isEnd: false}
     this.scroll = this.scroll.bind(this);
+    this.scrollAutomatically = this.scrollAutomatically.bind(this);
     this.list = this.props.listRef;
+    this.shouldScrollRight = true;
 
     this.setListRef = element => {
       this.list = element;
     };
+
+    if(this.props.shouldScrollAutomatically){
+      this.scrollAutomatically();
+    }
+  }
+
+  scrollAutomatically(){
+    const element = this.list;
+    if(!element){
+      setTimeout(() => this.scrollAutomatically(), 500);
+      return;
+    }
+    const scrollWidth = element.scrollWidth;
```

Comment:
I'm not a very big fan of having to “access” element globally. Can we make `scrollAutomatically` have to receive element as a parameter?

The issue with global variables is that you can't test the functions that access without mocking an entire context. When everything you need is given as parameter you can tell the next person that has to maintain that code that everything that's going to be used there is expected to be received there.

Reply:
So the Slider component receives a list that is horizontally scrollable. That is what `element` is. The thing is since I'm using references and `element` is a child of the Slider I can't access the reference right of the bat. I need to give `element` a function so itself can set the reference and Slider can use it to animate the scrolling. If I can use IDs then this should not be required, as I can just pass the ID as a prop to Slider. Did that make sense? Perhaps I'm missing a more obvious way to pass a reference to the Slider and can set this up in a different way.